streamer/src/nonblocking/quic.rs

```diff
@@ -557,6 +590,13 @@ pub mod test {
         let ip = "127.0.0.1".parse().unwrap();
         let server_address = s.local_addr().unwrap();
         let staked_nodes = Arc::new(RwLock::new(StakedNodes::default()));
+        if add_staked_peer {
```

Comment:
I think the tests would be clearer if they passed an optional `staked_nodes` instance to `setup_quic_server`